### PR TITLE
Feature/v1.0.0

### PR DESCRIPTION
**Pull request by codeclout**
same as feature/v0.13.0
major bump due to all updates not being backward compatible

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        this.host = this.options.host || process.env.HOST || '0.0.0.0';
+        this.name = this.options.name || pkg.name;
+        this.surrogateControlHeader = this.options.surrogateCacheControl || process.env.SURROGATE_CACHE_CONTROL || 'max-age=360, stale-while-revalidate=60, stale-if-error=86400';
+        this.contactEmail = this.options.contactEmail || undefined;
+        this.contactName = this.options.contactName || undefined;
+        this.contactUrl = this.options.contactUrl || undefined;
+        this.licenseName = this.options.licenseName || undefined;
+        this.licenseUrl = this.options.licenseUrl || undefined;
+        this.termsOfService = this.options.termsOfService || undefined;
+        this.maxListeners = this.options.maxListeners || 10;
```
Comment (randallbpotter15) [inline lib/config.js]: Do we want things like this "10" and (6*1000) from line 32 to live in a default constants file?
I'm always leery of having default values hard coded within the code.

Something like the below:
```
DEFAULT_MAX_LISTENERS = 10;
DEFAULT_METRICS_FLUSH_INTERVAL = (6 * 1000)
```

Comment (codeclout) [inline lib/config.js]: great idea!

Comment (codeclout) [inline lib/config.js]: @randallbpotter15 - PR has been updated with your feedback

Comment (randallbpotter15) [inline lib/config.js]: @codeclout  Looks good!  I think there were a few other places I saw that could use this.  Will take a look :)

Comment (randallbpotter15) [inline lib/config.js]: @codeclout Looked through - must have remembered wrong.  Rest of the way it's done looks good.

File: example/helpers/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+module.exports = {
```
Comment (adslaton) [inline example/helpers/index.js]: Add `use strict` statement.

**Comment by hhamilton-turner** [comment]
I sat with @codeclout , and he explained the changes to me. Approved.

@codeclout
Minor: Need new lines at the end of some of the files. Github points out which one specifically.

File: lib/registry.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        }
+    }
+
+    // require each plugin
+    get bundle() {
+        var obj,
+            plugin;
+
+        for (let b in this._map) {
+            obj  = Object.create(null),
```
Comment (adslaton) [inline lib/registry.js]: Change the comma to a semi-colon.

**Comment by adslaton** [comment]
All the changes are good and work as I expect. 

I am not convinced on the about the environment variable to being overridden by passed in options. This violates #3 of having 12 factor applications. See https://12factor.net/config.

I believe in order for this to move forward, the order should be changed to give priority over environment values.

**Pull request merged**